modules/plugins/languages/yaml.nix

```
@@ -0,0 +1,72 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}: let
+  inherit (builtins) attrNames;
+  inherit (lib.options) mkEnableOption mkOption;
+  inherit (lib.modules) mkIf mkMerge;
+  inherit (lib.lists) isList;
+  inherit (lib.types) enum either listOf package str;
+  inherit (lib.nvim.types) mkGrammarOption;
+  inherit (lib.nvim.lua) expToLua;
+
+  cfg = config.vim.languages.yaml;
+
+  defaultServer = "yaml-language-server";
+  servers = {
+    yaml-language-server = {
+      package = pkgs.nodePackages.yaml-language-server;
+      lspConfig = ''
+        lspconfig.yamlls.setup {
+          capabilities = capabilities;
+          on_attach = default_on_attach;
+          cmd = ${
+          if isList cfg.lsp.package
+          then expToLua cfg.lsp.package
+          else ''{"${cfg.lsp.package}/bin/yaml-language-server", "--stdio"}''
+        },
+        }
+      '';
+    };
+  };
+in {
+  options.vim.languages.yaml = {
+    enable = mkEnableOption "YAML language support";
+
+    treesitter = {
+      enable = mkEnableOption "Enable YAML treesitter";
```

```suggestion
      enable = mkEnableOption "YAML treesitter";
```

`mkEnableOption` already prefixes the description with "Whether to enable". Omit "Enable" here, and in `lsp.enable` below.